client/src/modules/journal/journal.js

```diff
@@ -277,6 +277,33 @@ function JournalController(Journal, Sorting, Grouping,
   // API register function
   function onRegisterApi(gridApi) {
     vm.gridApi = gridApi;
+    
+    vm.gridApi.edit.on.afterCellEdit($scope, function(rowEntity, colDef, newValue, oldValue){
+      if(newValue != oldValue) {
+        propagate(colDef.field,newValue);
+      }
+    });
+  }
+
+  function propagate(column, value){
+    var propagateColumn = ['trans_date', 'entity_uuid', 'origin_id'];
+    
+    //Check if the column updated must be propragated in all transaction
+    var checkPropagate = (propagateColumn.indexOf(column));
+
+    if(checkPropagate !== -1){
+      // For the old row
+      vm.transactions._entity.data.data.forEach(function (row) {
```

Comment:
If you move this function outside of the `propagate` function, it will reduce the number of memory allocations that are performed.   V8 will not garbage collect it as often.
```js
function updateSharedPropertyOnRow(row) {
 // function body
}

function propagate(column, value) {
  // function body
}
```